### PR TITLE
feat: inserting a question in a table

### PR DESCRIPTION
**Pull request by ykdojo**
This PR depends on #136 and #137.

Once those two and this one are merged, #108 will be closed.

### CODE REVIEW

**Comment by subhoghoshX** [comment]
I think you should open one PR at a time :rofl:  I'm confused between #137 and #138

**Comment by iShibi** [comment]
@ykdojo I think we can insert questions from front-end, this will simplify quite a lot of stuff. I'll open a PR later today.

**Comment by ykdojo** [comment]
@iShibi I actually explained my thought process behind this approach [here](https://github.com/ykdojo/supabase-nextjs-simple-auth). Mind taking a look? I think it's a nice, more simplified example than this PR.

File: src/pages/api/submit_question.js [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+import { createClient, SupabaseClient } from '@supabase/supabase-js';
+
+// TODO: move the following somewhere else.
+const supabaseUrl = process.env.NEXT_PUBLIC_SUPABASE_URL || '';
+const supabaseAnonKey = process.env.NEXT_PUBLIC_SUPABASE_ANON_KEY || '';
+const supabaseServiceKey = process.env.NEXT_PRIVATE_SUPABASE_SERVICE_KEY || '';
+const supabase = createClient(supabaseUrl, supabaseAnonKey);
+const supabaseSecret = createClient(supabaseUrl, supabaseServiceKey);
+
+export default async function handler(req, res) {
```
Comment (CerealPlayer) [inline src/pages/api/submit_question.js:11]:
```suggestion
export default async function handler(req: NextApiRequest, res: NextApiResponse) {
```
Always nice to have :)

File: src/pages/api/submit_question.js [hunk 2]
```diff
@@ -0,0 +1,41 @@
+// Next.js API route support: https://nextjs.org/docs/api-routes/introduction
+import { createClient, SupabaseClient } from '@supabase/supabase-js';
+
```
Comment (CerealPlayer) [inline src/pages/api/submit_question.js:3]:
```suggestion
import { NextApiRequest, NextApiResponse } from 'next';
```

**Review by CerealPlayer — CHANGES_REQUESTED** [review]

File: src/pages/submit.tsx [hunk 3]
```diff
@@ -97,21 +97,22 @@ const Submit: NextPage<UserProps> = ({ session }) => {
 									</div>
 
 									<div className='sm:col-span-3'>
-										<label htmlFor='recency' className='block text-sm font-medium text-gray-700'>
+										<label htmlFor='recency-weeks' className='block text-sm font-medium text-gray-700'>
```
Comment (debasishbsws) [inline src/pages/submit.tsx:100]:
```suggestion
										<label htmlFor='recency_weeks' className='block text-sm font-medium text-gray-700'>
```
change here also as you have done in the below fields.

File: src/pages/submit.tsx [hunk 4]
```diff
@@ -97,21 +97,22 @@ const Submit: NextPage<UserProps> = ({ session }) => {
...
 											How recently was this asked?
 										</label>
 										<div className='mt-1'>
 											<select
-												id='recency'
-												{...register('recency', { required: true })}
+												id='recency-weeks'
+												{...register('recency-weeks', { required: true })}
```
Comment (debasishbsws) [inline src/pages/submit.tsx:106]:
```suggestion
												id='recency_weeks'
												{...register('recency_weeks', { required: true })}
```

**Review by debasishbsws — COMMENTED** [review]
looks good

File: src/pages/submit.tsx (hunk 3 above)
Comment (ykdojo) [reply src/pages/submit.tsx:100]: @debasishbsws thank you but this is an old PR so I'll close it

**Pull request closed without merging**